### ISSUE TITLE
Bug: @typescript-eslint/no-deprecated Does not recognize subnamespace

### ISSUE THREAD

**Issue by xuhdev** [typescript-eslint/typescript-eslint#10220]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.6.2&fileType=.tsx&code=PQKhCgAIUgBATApgBwE6IMYEMAuj6QDOyWGiAdFpAJaGRJqa77lQjDhIYA2W6kAOywBbRMVKIiJMpUgBvcAF9wnTL35DR4slInkARvKg0BeVADMJkAJKnEFqwsjP6ALiI5U1AQHMA3MbKyuDAwJAAKgAWkuYA9tzcsQDu3j6Q3N6ShJGxAK7c8AIA5DiQ6MixqKVYApD2qJXgGLEChKUAbu7aFPrktmaWOgC88vDuRZHURcpAA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oBNFjpEZAIb5E3dFETRoHaJHBgAviEVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
/**
 * @deprecated space.a is deprecated.
 */
declare namespace space.a {
}

declare namespace space.b {
  interface Interface {
    d: string;
  }
}

// The following line shouldn't report an error
const v: space.b.Interface = {d: 'hi'}
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-deprecated": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

`const v: space.b.Interface = {d: 'hi'}` shouldn't report an error

### Actual Result

`const v: space.b.Interface = {d: 'hi'}` reports a `@typescript-eslint/no-deprecated` error

**Comment by bradzacher** [comment]
This is intentional really - your code is ambiguous. Is it meant to mean:
- JUST `space`, or
- JUST `space.a` or
- `space` AND `space.a`

It's impossible to tell really.

Instead use the non-nested syntax which disambiguates your meaning so we can accurately understand which symbol should be deprecated.
```ts
declare namespace space {
  /**
   * @deprecated space.a is deprecated.
   */
  export namespace a { }
}
```
[playground](https://typescript-eslint.io/play/#ts=5.6.2&fileType=.tsx&code=FAEwpgxgNghgTmABAOxgWzAZwA4wknPJAb2EUQHoAqKs8qxAAXGwQhgBcwRFD8A6GIgCWmRCzadu-OoioU6YAB7YA9nA4p0WXPkRDiiAL7AToSLARaMfArrD8ARolLlhyLnABmRRAEkPMG9fV3JxAC5eDjh3AHMAbjoTMwoKRAAVAAskL1UoKFUAdzjEKHcCTNUAVygQZAByTQQ1DX1kRCC4dWAIVWRMTQA3SNsnfgDPHz0AXhcQSPrM4XqTIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oBNFjpEZAIb5E3dFETRoHaJHBgAviEVA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)
